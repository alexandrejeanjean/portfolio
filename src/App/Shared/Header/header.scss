.header {
  position: fixed;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90vw;
  cursor: pointer;
  background-color: transparent;
  text-transform: uppercase;
  padding-top: 20px;
  padding-left: 5vw;
  z-index: 10;
  .author-name {
    font-family: 'NeutraTextBook', 'Arial';
    font-size: 0.6em;
  }

  .contact {
    width: 40px;
    animation-name: gelatine;
    animation-delay: 0.4s;
    animation-duration: 0.8s;
    animation-iteration-count: 2;
  }
}

@media screen and (max-width: 768px) {
  .header {
    width: 85vw;
    height: 20px;
  }
}

@keyframes gelatine {
  from,
  to {
    transform: scale(1, 1);
  }
  25% {
    transform: scale(0.9, 1.1);
  }
  50% {
    transform: scale(1.1, 0.9);
  }
  75% {
    transform: scale(0.95, 1.05);
  }
}
