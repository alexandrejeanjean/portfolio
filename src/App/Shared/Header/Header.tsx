import React from "react";
import { Link } from "react-router-dom";

import { Mail } from "../../Assets/imgPath";
import "./header.scss";

const Header = () => (
  <header className="header">
    <Link className="author-name" to="/">
      alexandre jeanjean
    </Link>
    <a href="mailto:jeanjeanalexandre@gmail.com" title="contactez-moi">
      <img className="contact" src={Mail} alt="" />
    </a>
  </header>
);

export default Header;
