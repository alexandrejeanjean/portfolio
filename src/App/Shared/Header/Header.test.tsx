import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header', () => {
  it('should render header correctly', () => {
    const header = shallow(<Header />)
    expect(header).toMatchSnapshot()
  })
})
