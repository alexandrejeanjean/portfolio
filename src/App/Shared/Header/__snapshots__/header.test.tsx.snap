// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header correctly 1`] = `
<header
  className="header"
>
  <Link
    className="author-name"
    to="/"
  >
    alexandre jeanjean
  </Link>
  <a
    href="mailto:jeanjeanalexandre@gmail.com"
    title="contactez-moi"
  >
    <img
      alt=""
      className="contact"
      src={[Function]}
    />
  </a>
</header>
`;
